Change some blockitem entries

KlemenDEV opened the pull request:
Was an attempt to resolve #3998, but did not succeed as there is _no tag for concrete and concrete_powder_.

We can remove universal variants of those instead, or leave it as it is (open to discussion). I would prefer to keep it as is in case forge or some other modding API add such tags in the future and also for backwards compatibility.

However, PR is here as I changed some other entries while checking this.

AlsoSomeoneElse commented:
This PR seems to break the block list in features, which expects a block id and not a tag. Here dirt becomes `minecraft:air` instead of `minecraft:dirt`
![issue](https://github.com/MCreator/MCreator/assets/64546452/95af1e15-60cc-4f49-8fa8-197e62d621a3)

Another issue I found while testing, it seems some jolly types aren't marked properly. Again here in the block selector the "Any log" appears, but not the "Any sand"
![issue 2](https://github.com/MCreator/MCreator/assets/64546452/1ce9701c-c515-4093-a4c9-d9d0465e6ba3)

KlemenDEV commented:
Thank you for the review.

I believe this is because logs don't map to tags actually, but to the actual value, while eg. sand maps to tag, so it has subtypes marked.

I will keep things as they are currently in this case to not break more things than fix as this PR was intended to fix the open issue which it does not. In other words, system as is is not optimal, but works so I won't break what works.

But we will need to rework this list in the future possibly to be better organized. However I don't think the time is now so am closing this PR for now.

KlemenDEV commented:
So updated changes are:

* added wildcard variant to some entries that did not have it
* use vanilla tags when possible
* removed wildcard concrete and concrete_powder from the lists as they don't actually exist

There are probably more issues on the data list to be fixed, but if this PR does not make things worse and instead fixes at least some of them, I think it is a good step forward.

Goldorion commented on plugins/mcreator-core/datalists/blocksitems.yaml:
```diff
@@ -3710,12 +3709,6 @@
   type: block
   texture: CONCRETE_POWDER#15
 
-- Blocks.CONCRETE:
```
The changes for MC's tags are fine (and the FB Generator thanks you!). However, concerning the 2 removed entries... not too sure it's a good idea. See the image
![image](https://github.com/MCreator/MCreator/assets/38427877/0e8ed9ac-162d-4873-8f61-a7b08a6dac0a)

The missing entries are converted to air... it doesn't break workspaces, but I can already some potential issues:
1. My ReCiPeS (or whatever it is) ArE WeIrD! HELP ME PLEASE!!!
2. This might break break some custom logic (mainly in procedures I think) made by users.

However, as they weren't real wildcards, this might be easily fix by automatically change them to White concrete (powder), so users don't notice.

KlemenDEV replied:
I thought to just remove them because they don't work the way users would want them to, so even if we convert them to white concrete, it won't work as "any concrete" as the user may anticipate.

The only place where they could use them so far was in recipes because it had "subtypes" and thus did not load when tags were not allowed

Goldorion reviewed (approved):
The previous issue I reported seems to be fine, so the PR is good for me

Defeatomizer reviewed (approved):
I think this is fine.

KlemenDEV commented:
Thank you for the reviews!
